app/server_test.go

```diff
@@ -33,10 +33,16 @@ import (
 )
 
 func TestStartServerSuccess(t *testing.T) {
-	s, err := NewServer()
+	configStore, _ := config.NewMemoryStore()
+	store, _ := config.NewStoreFromBacking(configStore, nil, false)
+	cfg := store.Get()
+	*cfg.ServiceSettings.ListenAddress = ":0"
+
+	store.Set(cfg)
+
+	s, err := NewServer(ConfigStore(store))
```

Comment:
Would it make sense to turn this into a function? Something like `newServerWithConfig`.